### ISSUE TITLE
setRepository() in RepositoryItemReader in 5.0.0-M4 broken

### ISSUE THREAD

**Issue by makigumo** [spring-projects/spring-batch#4164]
**Bug description**
With https://github.com/spring-projects/spring-batch/commit/a30aaac4be6bdf66d21a252c87dd5d15b1239bd1 in [5.0.0-M4](https://github.com/spring-projects/spring-batch/releases/tag/5.0.0-M4) it is no longer possible to use queries in `RepositoryItemReader()` or `RepositoryItemReaderBuilder` that return a different type than the repository's entity type.

**Environment**
JKD 17.0.3, Spring Boot 3.0.0-M4

**Steps to reproduce**
Update from Spring Boot 3.0.0-M3 to 3.0.0-M4.

**Expected behavior**
Same behavior as in .

**Minimal Complete Reproducible example**

Simplified illustrative example:
Consider something like the following repository.
```java
public interface TestRepository extends JpaRepository<TestEntity, Long> {

    @Query(value = "SELECT e.id FROM TestEntity e")
    Page<String> getFromCustomQuery(Pageable pageable);
}
```

With the following batch configuration.
```java
@Configuration
public class TestConfig {

    private final TestRepository testRepository;

    @Bean
    @StepScope
    protected RepositoryItemReader<String> testItemReader() throws Exception {
        final var itemReader = new RepositoryItemReader<String>();
        itemReader.setRepository(testRepository);
        itemReader.setPageSize(1000);
        itemReader.setSort(Map.of("id", Sort.Direction.ASC));
        itemReader.setMethodName("getFromCustomQuery");
        itemReader.afterPropertiesSet();
        return itemReader;
    }
}
```

**Comment by makigumo** [comment]
Related/Repeated issue: https://github.com/spring-projects/spring-batch/issues/1369

**Comment by fmbenhassine** [comment]
Thank you for raising this! I think the generic type was mistakenly changed as in: https://github.com/spring-projects/spring-batch/commit/b26d272d436d77bb6b4d0e7320387351937a1646#diff-268ec158a27f386faec50b4fff70ad0fe62c54b4a11cb6a1b146b789b4584d87R68. Will fix that in 5.0 M5.
